app.rb

```diff
@@ -1,8 +1,70 @@
 require 'sinatra/base'
+require 'sinatra/reloader'
+
 class RockPaperScissors < Sinatra::Base
+  configure :development do
+    register Sinatra::Reloader
+  end
+
+  enable :sessions
+
   get '/test' do
     'test page'
   end
 
+  get '/' do
+    erb(:index)
+  end
+
+  post '/name' do
+    session[:player1] = params[:player1]
+    redirect('/game')
+  end
+
+  get '/game' do
+    @player1 = session[:player1]
+    erb(:game)
+  end
+
+  post '/weapon' do
+    session[:weapon] = params[:weapon]
+    redirect('/result')
+  end
+
+  def computer_weapon
+    ['Rock', 'Paper', 'Scissors'].sample
+  end
+
+  get '/result' do
+    @weapon1 = session[:weapon]
+    @weapon2 = computer_weapon
+    if @weapon1 == @weapon2
+      @message = "It's a draw!"
```

Comment:
as before, might be better to have game logic extracted from your controller. Additionally in this case, this pathing now violates SRP pretty hard ;)